src/main/java/org/ibp/api/java/impl/middleware/germplasm/validator/GermplasmNameValidator.java

```diff
@@ -0,0 +1,147 @@
+		if(germplasmNameRequestDto.isPreferredName() != null){
+			errors.reject("germplasm.name.preferred.required", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+	}
+
+	public void validateDeleteName(final GermplasmNameRequestDto germplasmNameRequestDto) {
+		this.errors = new MapBindingResult(new HashMap<String, String>(), GermplasmNameRequestDto.class.getName());
+		germplasmValidator.validateGermplasmId(errors, germplasmNameRequestDto.getGid());
+
+		if (errors.hasErrors()) {
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+
+		final Name name = germplasmNameService.getNameByNameId(germplasmNameRequestDto.getId());
+		this.ValidateGermplasmName(errors, germplasmNameRequestDto, name);
+		this.ValidateDeletepreferredName(errors, name);
+
+	}
+
+	protected void ValidateGermplasmName(final BindingResult errors, final GermplasmNameRequestDto germplasmNameRequestDto,
+		final Name name) {
+		if (name == null || !name.getGermplasmId().equals(germplasmNameRequestDto.getGid())) {
+			errors.reject("germplasm.name.invalid", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+
+		}
+	}
+
+	protected void ValidateDeletepreferredName(final BindingResult errors, final Name name) {
+		if (name.getNstat().equals(1)) {
+			errors.reject("germplasm.name.preferred.invalid", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+
+		}
+	}
+
+	protected void validateUpdatePreferredName(final BindingResult errors, final GermplasmNameRequestDto germplasmNameRequestDto,
+		final Name name) {
+		if (!germplasmNameRequestDto.isPreferredName() && name.getNstat().equals(1)) {
+			errors.reject("germplasm.name.preferred.invalid", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+		}
+	}
+
+	protected void validateNameLength(final BindingResult errors, final GermplasmNameRequestDto germplasmNameRequestDto) {
+		if (StringUtils.isBlank(germplasmNameRequestDto.getName())) {
+			errors.reject("germplasm.name.required", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+
+		}
+
+		if (germplasmNameRequestDto.getName().length() > LOT_NOTES_MAX_LENGTH) {
+			errors.reject("germplasm.name.length", "");
+			throw new ApiRequestValidationException(errors.getAllErrors());
+
+		}
+	}
+
+	protected void validateNameDate(final BindingResult errors, final GermplasmNameRequestDto germplasmNameRequestDto) {
+		if (germplasmNameRequestDto.getDate() == null) {
+			errors.reject("germplasm.name.date.required", "");
+		}
+
+		try {
+			Util.getSimpleDateFormat(Util.DATE_AS_NUMBER_FORMAT).parse(germplasmNameRequestDto.getDate().toString()); // verificar esto.
+		} catch (ParseException e) {
+			errors.reject("germplasm.name.date.invalid", new Object[] {
+					germplasmNameRequestDto.getDate().toString()},
+				"Invalid date value found.");
```

Comment:
not sure why you need to send the "Invalid date value found" string if there is only one row and date in the request?